### DISCUSSION TITLE
Conditional response options for multiple choice question

### DISCUSSION THREAD

**Discussion by isabellesander** [jspsych/jsPsych#1206] [Q&A]
Hi!

I am trying to create a discrete choice experiment and have a question regarding the survey-multi-choice plug-in option.
I want to display 3 images of robots and ask participants to answer the two questions which one they like best and which one they like least of the three.
I have created a survey-multi-choice trial with the three pictures as prompts without answer options as well as two questions as prompts ("which one is most suitable" and "which one is least suitable"). These 2 questions each have answering options "robot category 1", "robot category 2", and "robot category 3" (referring to the three pictures).
I am now wondering if there is an option to make sure that if robot category 1 is selected for most suitable it cannot be chosen in the second question as least suitable. 
I am having to use this for my bachelor thesis and am unfortunately an absolute beginner in coding.
If this has been answered elsewhere I would be thankful for a pointer in that direction.

Here is what I have so far:

```js
      var scale_options = ["Roboterkategorie 1","Roboterkategorie 2","Roboterkategorie 3"];
      var multi_choice_block_horizontal = {
    type: 'survey-multi-choice',
    questions: [
      {prompt: '<img src="img/1.png" style="width:288px;height:400px; border: 5px solid #999999;" alt="1" />                       <img src="img/2.png" style="width:288px;height:400px;border: 5px solid #999999" alt="2" />       <img src="img/3.png" style="width:288px;height:400px;border: 5px solid #999999" alt="3"/>',options: false, required: false},
              {prompt: "<strong>Welche der Robotervarianten ist f&uuml;r den Einsatz im Pflegekontext am <u>besten</u> geeignet?</strong>",   options:scale_options, required: true, horizontal: true},
              {prompt: "<strong>Welche der Robotervarianten ist f&uuml;r den Einsatz im Pflegekontext am <u>schlechtesten</u> geeignet?</strong>",   options:scale_options, required: true, horizontal: true}
    ],
    preamble: '<p><i> W&auml;hlen Sie bitte spontan aus, welche Roboterkategorie Ihrer Meinung nach am <strong>besten</strong> und welche am <strong>schlechtesten</strong> f&uuml;r sowohl <strong>soziale</strong> als auch <strong>funktionale</strong> Aufgaben im Pflegekontext geeignet ist. </i></p>',
    button_label: 'Weiter',
  };
```

Additionally I am wondering why the 3 pictures and the options aren't centered on screen. 

Thank you so much for your help and mercy with an absolute newbie!
Isabelle

**Comment by jodeleeuw** [comment]
Hi Isabelle,

> Additionally I am wondering why the 3 pictures and the options aren't centered on screen.

Can you share a screenshot of what it looks like when you run the experiment? (You can drag and drop and image right into the text editor here to add it to your post).

One idea that comes to mind is that you could move the images of the three robots into the `preamble` parameter, instead of using a question to show them. That might help with some formatting issues. You should be able to just copy the prompt from question 1 and add it to the end of the preamble.

Ensuring that participants don't pick the same option for both questions will be harder. The plugin wasn't developed with this application in mind. I wonder if the [maxdiff plugin that we are adding to 6.2](https://github.com/jspsych/jsPsych/pull/1136) will be a better fit for this task? It is possible to do with the current plugin, but I think it will involve create a [looping timeline](https://www.jspsych.org/overview/timeline/#looping-timelines) where you repeat the trial if the participant selected the same answer twice.

**Reply by isabellesander** [reply]
Thanks so much for the quick reply.
I will see if putting the pictures in the preamble is an option. 
But that still doesn't help with the response options I guess. Here is a screenshot of what the current version looks like: 

<img width="1416" alt="Screen Shot 2020-11-20 at 3 20 32 PM" src="https://user-images.githubusercontent.com/74778488/99812818-319d5900-2b47-11eb-8f81-1e8b3715f28a.png">

As you can see, the questions are centered but the response options are aligned left. (The preamble is cut off in this screenshot but is also centered).

**Comment by jodeleeuw** [comment]
@becky-gilbert, one idea I had from this thread is that maybe when we overhaul the survey plugins in v7.1 we should add a `validation` parameter to each of them so that this kind of behavior is easier to achieve.

**Reply by becky-gilbert** [reply]
Agree! I opened an issue for notes on the survey plugin overhaul and included your suggestion for a `validation` parameter: #1207
